docs/dev/Architecture.md

```diff
@@ -0,0 +1,39 @@
+# OpenDistro SQL Engine Architecture
+
+---
+## 1.Overview
+
+The OpenDistro SQL (OD-SQL) project is developed based on NLPChina project (https://github.com/NLPchina/elasticsearch-sql) which has been deprecated now ([attributions](https://github.com/opendistro-for-elasticsearch/sql/blob/master/docs/attributions.md)). Over the one year in development, a lot of features have been added to the OD-SQL project on top of the existing older nlp-china project. The purpose of this document is to explain the OD-SQL current architecture going ahead.
```

Comment:
np: should we maintain consistency with how we refer to NLPChina? e.g NLPChina vs nlp-china

Reply:
Good catch. Will stick with NLPChina. Thanks!